### ISSUE TITLE
False FileNotFound thrown by SSM

### ISSUE THREAD

**Issue by aurorahunter** [Intel-bigdata/SSM#1620]
My Rule:
file: every 100000m | path matches "/tmp/mar7/*" | copy2s3 -dest "s3a://dummy" ; truncate0

Directory structure of /tmp/mar7 is

```
ssm_deploy_eq_st@musstg050111:~/MAR7/smart-data-1.4.0-SNAPSHOT$ hdfs dfs -ls -R /tmp/mar7
drwxr-xr-x   - ssm_deploy_eq_st supergroup          0 2018-03-07 23:37 /tmp/mar7/tmp3
drwxr-xr-x   - ssm_deploy_eq_st supergroup          0 2018-03-07 23:37 /tmp/mar7/tmp3/tmp2
-rw-r--r--   3 ssm_deploy_eq_st supergroup 2097258880 2018-03-07 23:37 /tmp/mar7/tmp3/tmp2/V714679_MDF4_20171020T123100-20171020T123230.MF4
-rw-r--r--   3 ssm_deploy_eq_st supergroup 2097259408 2018-03-07 23:37 /tmp/mar7/tmp3/tmp2/V714679_MDF4_20171020T123529-20171020T123700.MF4
-rw-r--r--   3 ssm_deploy_eq_st supergroup 2097258960 2018-03-07 23:37 /tmp/mar7/tmp3/tmp2/V714679_MDF4_20171020T124438-20171020T124609.MF4
```

The dir /tmp/mar7/ has multiple sub directories and SSM tries to schedule A COPY2S3 action for directories and obviously results in false FileNotFound .

```
java.io.FileNotFoundException: Path is not a file: /tmp/mar7/tmp3
        at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:70)
        at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:56)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocationsInt(FSNamesystem.java:2038)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:2008)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:1921)
        at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getBlockLocations(NameNodeRpcServer.java:572)
        at org.apache.hadoop.hdfs.server.namenode.AuthorizationProviderProxyClientProtocol.getBlockLocations(AuthorizationProviderProxyClientProtocol.java:89)
        at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getBlockLocations(ClientNamenodeProtocolServerSideTranslatorPB.java:365)
        at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
        at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:617)
        at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1073)
        at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2217)
        at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2213)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:422)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1917)
        at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2211)

        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.apache.hadoop.ipc.RemoteException.instantiateException(RemoteException.java:106)
        at org.apache.hadoop.ipc.RemoteException.unwrapRemoteException(RemoteException.java:73)
        at org.apache.hadoop.hdfs.DFSClient.callGetBlockLocations(DFSClient.java:1281)
        at org.apache.hadoop.hdfs.DFSClient.getLocatedBlocks(DFSClient.java:1266)
        at org.apache.hadoop.hdfs.DFSClient.getLocatedBlocks(DFSClient.java:1254)
        at org.apache.hadoop.hdfs.DFSInputStream.fetchLocatedBlocksAndGetLastBlockLength(DFSInputStream.java:305)
        at org.apache.hadoop.hdfs.DFSInputStream.openInfo(DFSInputStream.java:271)
        at org.apache.hadoop.hdfs.DFSInputStream.<init>(DFSInputStream.java:263)
        at org.apache.hadoop.hdfs.DFSClient.open(DFSClient.java:1585)
        at org.apache.hadoop.hdfs.DistributedFileSystem$4.doCall(DistributedFileSystem.java:326)
        at org.apache.hadoop.hdfs.DistributedFileSystem$4.doCall(DistributedFileSystem.java:322)
        at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
        at org.apache.hadoop.hdfs.DistributedFileSystem.open(DistributedFileSystem.java:322)
        at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:783)
        at org.smartdata.hdfs.action.Copy2S3Action.getSrcInputStream(Copy2S3Action.java:221)
        at org.smartdata.hdfs.action.Copy2S3Action.copySingleFile(Copy2S3Action.java:189)
        at org.smartdata.hdfs.action.Copy2S3Action.execute(Copy2S3Action.java:144)
        at org.smartdata.action.SmartAction.run(SmartAction.java:124)
        at org.smartdata.server.engine.cmdlet.Cmdlet.runAllActions(Cmdlet.java:94)
        at org.smartdata.server.engine.cmdlet.Cmdlet.run(Cmdlet.java:108)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by qiyuangong** [comment]
Hi @aurorahunter. Thank you. I'm working on this issue. :)

**Comment by qiyuangong** [comment]
This issue is fixed with #1621 . Please check it.

**Comment by qiyuangong** [comment]
Issue closed. Feel free to re-open it, if there is any problem.
